packages/editor/src/components/browser/DataTypeBrowser.test.tsx

```diff
@@ -0,0 +1,103 @@
+import { TableUtil, render, screen, userEvent } from 'test-utils';
+import { useDataTypeBrowser } from './DataTypeBrowser';
+
+const Browser = (props: { location: string; accept: (value: string) => void }) => {
+  const browser = useDataTypeBrowser();
+  return (
+    <>
+      {browser.content}
+      <button data-testid='accept' onClick={() => props.accept(browser.accept())} />
+    </>
+  );
+};
+
+describe('DataTypeBrowser', () => {
+  function renderBrowser(options?: { location?: string; accept?: (value: string) => void }) {
+    render(<Browser location={options?.location ?? 'something'} accept={options?.accept ?? (() => {})} />, {
+      wrapperProps: {
+        meta: {
+          datatypes: [
+            {
+              simpleName: 'Person',
+              packageName: 'ch.ivyteam.test',
+              fullQualifiedName: 'ch.ivyteam.test.Person'
+            }
+          ],
+          dataClasses: [
+            {
+              name: 'AddContactData',
+              fullQualifiedName: 'ch.ivyteam.documentation.project.AddContactData',
+              packageName: 'ch.ivyteam.documentation.project',
+              path: 'dataclasses/ch/ivyteam/documentation/project/AddContactData.ivyClass'
+            },
+            {
+              name: 'Person',
+              fullQualifiedName: 'ch.ivyteam.test.Person',
+              packageName: 'ch.ivyteam.test',
+              path: 'dataclasses/ch/ivyteam/test/Person.ivyClass'
+            },
+            {
+              name: 'List',
+              packageName: 'java.util',
+              fullQualifiedName: 'java.util.List',
+              path: 'thisisaTest'
+            }
+          ]
+        }
+      }
+    });
+  }
+
+  test('renderOnlyDataClasses', async () => {
+    renderBrowser();
+    await TableUtil.assertRowCount(3);
+  });
+
+  test('renderCombinedDatatype', async () => {
+    renderBrowser();
+    await userEvent.click(await screen.findByPlaceholderText('Search'));
+    await userEvent.keyboard('Person');
+    screen.debug();
+    await TableUtil.assertRowCount(1);
+  });
+
+  test('accept', async () => {
+    let data = '';
+    renderBrowser({ accept: value => (data = value) });
+    await userEvent.click(await screen.findByRole('cell', { name: 'AddContactData' }));
+    await userEvent.click(screen.getByTestId('accept'));
```

Comment:
I used 'testid' because it is already used in the testing of the Attribute Browser and the CMS Browser. But I can change it if you prefer.

Reply:
I would only use the testId if nothing else is working because the testid will be included in the final build product, this means test code inside of the product. And if there is no big reason for this, this should be avoided :)